web-ui/src/main/resources/catalog/components/utility/CORSInterceptor.js

```diff
@@ -58,6 +60,28 @@
                   (config.url.indexOf('http') !== 0 &&
                   config.url.indexOf('//') !== 0);
 
+              // If this is an authorized mapservice then we need to adjust the url or add auth headers
+              // Only check http url and exclude any urls like data: which should not be changed. Also, there is not need to check prox urls.
+              if (config.url !== null && config.url.startsWith("http") && !config.url.startsWith(gnGlobalSettings.proxyUrl)) {
+                 var mapservice = gnMapServicesCache.getMapservice(config.url);
+                 if (mapservice !== null) {
+                    if (mapservice.useProxy) {
+                       // If we are using a proxy then add it to the list of required proxies.
+                       if ($.inArray(url, gnGlobalSettings.requireProxy) === -1) {
+                         var url = config.url.split('/');
+                         url = url[0] + '/' + url[1] + '/' + url[2] + '/';
```

Comment:
Can you explain this code?

Reply:
I copied it from the following

https://github.com/geonetwork/core-geonetwork/blob/1b8c4fc1b7b4be7f55140276d62b27f5507fd4dc/web-ui/src/main/resources/catalog/components/utility/CORSInterceptor.js#L140-L147

It seems like a weird way to get the url in the format of  "http://domain/"